### PR TITLE
Rewrite to traverse the code

### PR DESCRIPTION
**Pull request by chopmo**
It sort of works...I think: 

image

### CODE REVIEW

**Comment by chopmo** [comment]
@tgk Oh, I actually meant to push this PR to my own fork of the project...

File: src/plait/core.clj [hunk 1]
```diff
@@ -25,15 +25,21 @@
...
+(defn plait-impl [bindings x]
+  (if (and (seqable? x)
+           (= 'plait (first x)))
+    (let [child-bindings (rename-underscore-bindings (second x))
+          child-body (drop 2 x)
+          new-bindings (merge-bindings bindings child-bindings)]
+      (plait-impl new-bindings
+                  (concat ['let new-bindings]
+                          (map (partial plait-impl new-bindings) child-body))))
+    x))
```
Comment (tgk) [inline src/plait/core.clj]: Hmm I think this breaks if you do this:

```
(plait [a 1
          b (inc a)]
  (do
    (plait [a 2]
      b)))
```

as it does not recur on the `do`?

**Review by tgk — COMMENTED** [review]
Nice! I think we should test it on our own codebase :wink:

**Review by tgk — COMMENTED** [review]
Nice! I'm starting to think maybe walking would be better :thinking:

**Comment by chopmo** [comment]
@tgk I'm getting into adding a lot of special cases... I'll need a more general solution. I thought I could use `(into (empty x..` but it breaks. The current version works on some of our non-trivial test files though!

**Comment by tgk** [comment]
Nice. I would guess the `clojure.walk` walkers would have a nicer solution for this.

**Comment by chopmo** [comment]
@tgk Hm. I finally found a solution based on `walk` that I was really happy with, but now the line numbers are broken again when a test fails :( I don't understand why. I'm not using macroexpand anymore.

**Comment by chopmo** [comment]
Yikes, I'm confused. I can reproduce with a very small example. But it looks to me like the walk and recursive versions produce exactly the same code. This is one thing I'd like to bring to a Friday mentoring session if you're up for it :)

**Pull request merged**